ttnn/cpp/ttnn/tensor/tensor_ops.cpp

```diff
@@ -391,7 +391,12 @@ Tensor tensor_reshape(const Tensor& input_tensor, const ttnn::Shape& new_shape)
                     if (tensor.memory_config().memory_layout != TensorMemoryLayout::HEIGHT_SHARDED) {
                         DeviceStorage device_storage = std::get<T>(tensor.get_storage());
                         DeviceBuffer device_buffer = device_storage.get_buffer();
-                        device_buffer->set_page_size(new_shape[-1] * tensor.element_size());
```

Comment:
Please replace with 
```
const auto& tensor_spec = tensor.tensor_spec();
auto page_size_bytes = tensor_spec.compute_page_size_bytes();
device_buffer->set_page_size(page_size_bytes);
```

@sminakov-tt , right?

Reply:
Yes, I think this should be the correct way

Reply:
okok, I replaced with `compute_page_size_bytes`